pyproject.toml

```diff
@@ -43,6 +43,7 @@ dependencies = [
     "pydantic>=2.7.4,<3.0",
     "docstring-parser>=0.15,<1.0",
     "jiter>=0.5.0",
+    "typing-extensions>=4.10.0",
```

Comment:
wait is `typing-extensions` not a built-in package?

also, after looking into it briefly I'm wondering if we even need the extension since we only support 3.10+? are there issues with just importing things like `TypedDict` directly from typing? I don't think there are 3.11+ features we're using?

Reply:
`typing-extensions` is a non-build-in, optional typing backport.
Since `pydantic` already depends on `typing-extensions`, we do not expect to add any new dependencies.
Changed from `TypeGuard` to `TypeIs` for stricter type refinement.
This is a type feature from 3.13 and requires `typing-extensions`.
https://peps.python.org/pep-0742/